### PR TITLE
[RFC] add new libtls implementation

### PR DESCRIPTION
**Pull request by ericonr**
Based on #21056 and some discussion in #20935 

@leahneukirchen has suggested an alternative that is building only `libtls` from LibreSSL instead. I haven't tried this one yet.( See https://github.com/libressl-portable/portable/commit/fe42a8011b25797afa48cf0359c7e2753b2ece5f )

My current propositions are:

- libretls from causal agency, who is the author of most of the software in our repos that depend on libtls
- libtls-bearssl, which is based on BearSSL instead of any sort of mix between LibreSSL/OpenSSL

### CODE REVIEW

File: srcpkgs/libretls/template [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Template file for 'libretls'
+pkgname=libretls
+version=3.3.1
+revision=1
+build_style=gnu-configure
+makedepends="openssl-devel"
```
Comment (Johnnynator) [inline srcpkgs/libretls/template:6]: Misses `pkg-config` in `hostmakedepends`

**Comment by ericonr** [comment]
The `libtls-standalone` option is https://gitlab.alpinelinux.org/alpine/aports/-/blob/6012e47e379b125867e0dddada4ce1af2d589d5a/main/libtls-standalone/APKBUILD , but it is unmaintained (was only moved back to main because other things depend on it), and idk if all these workarounds are necessary with recent LibreSSL.

**Comment by leahneukirchen** [comment]
LibreSSL libtls: #28732

**Comment by ericonr** [comment]
Closing in favor of leah's approach.

**Pull request closed without merging**